### PR TITLE
Add Booksidian plugin

### PR DESCRIPTION
**Pull request by Morglaf**
# I am submitting a new Community Plugin

## Repo URL
https://github.com/Morglaf/booksidian

Link to my plugin: https://github.com/Morglaf/booksidian

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [ ] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by joethei** [comment]
Closing as duplicate of #3820

**Comment by joethei** [comment]
["name": "ObsiBook",](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/manifest.json#L3)
Please don't include "Obsidian" or any abbreviations of it in the plugin name.

["description": "Creating books with latex, Pandoc, MiKTeX, pdftk.",](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/manifest.json#L6)
Please make sure that you are using the correct cassing for these names, for example, it should be "LaTeX"
Also, "Create" would be the better word.

[const VIEW_TYPE_OBSIBOOK = "obsibook-view";](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L40)
Don't add the `view` suffix.

[containerEl.createEl('label', { text: 'Chemin d\'exportation :' });](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L146), [.setName('Conserver le dossier temporaire')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L168), [containerEl.createEl('label', { text: 'Sélectionner une couverture :' });](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L192), [.setName('Épaisseur de la tranche')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L212-L213), [const generateCoverButton = containerEl.createEl('button', { text: 'Générer la couverture' });](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L223), [this.dynamicFieldsContainerCover.createEl('label', { text: 'Champs dynamiques détectés de la couverture :' });](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L242), [this.dynamicFieldsContainerCover.createEl('span', { text: 'Aucun champ dynamique détecté.' });](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L247), [.setName('Épaisseur du papier')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L1095), [.setName('Chemin de couverture')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L1108) and more
Some of the text in this plugin is in english, some is in french, please stick to one language or add a i18n provider.

[this.app.workspace.getLeavesOfType(VIEW_TYPE_OBSIBOOK).forEach(leaf => leaf.detach());](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L407)
Do not detach leaves with your custom view in `onunload`, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>

[containerEl.createEl('h2', { text: 'Settings for ObsiBook plugin' });](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L994)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[.setName('Pandoc Path')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L997), [.setName('Template Folder Path')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L1008), [.setName('LaTeX Template')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L1021), [.setName('Xelatex Path')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L1035), [.setName('Output Folder Path')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L1046), [.setName('Imposition Path')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L1057), [.setName('Keep Temp Folder')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L1072)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[.setName('Xelatex Path')](https://github.com/Morglaf/ObsiBook/blob/b6c40fa5d70413894b86713b5abddcd6f0c28ebd/main.ts#L1035)
It's called "XeLaTeX"
